src/machine.rs

```diff
@@ -639,15 +637,20 @@ impl<'tcx> MiriMachine<'tcx> {
         });
         let rng = StdRng::seed_from_u64(config.seed.unwrap_or(0));
         let borrow_tracker = config.borrow_tracker.map(|bt| bt.instantiate_global_state(config));
-        let data_race = match &config.data_race_detection {
-            DataRaceConfig::None => DataRaceHandler::None,
-            DataRaceConfig::DataRace { weak_memory_emulation } =>
-                DataRaceHandler::DataRace(Box::new(data_race::GlobalState::new(
-                    config,
-                    *weak_memory_emulation,
-                ))),
-            DataRaceConfig::Genmc(_genmc_config) => DataRaceHandler::Genmc(genmc_ctx.unwrap()),
-        };
+        let data_race =
+            match (config.data_race_detector, config.weak_memory_emulation, &config.genmc_config) {
+                (true, weak_memory_emulation, None) =>
```

Comment:
This is super hard to read. An `if let Some(genmc_cfg) = config.genmc_config` would be a lot easier to read I think.